guvnor-webapp/src/main/java/org/guvnor/client/perspectives/M2RepoPerspective.java

```diff
@@ -61,23 +58,16 @@ private void init() {
         add( m2RepoEditor );
     }
 
+    @Inject
+    UploadFormPresenter uploadFormPresenter;
+
     @WorkbenchMenu
     public Menus getMenus() {
         return MenuFactory.newTopLevelMenu( AppConstants.INSTANCE.Upload() )
                 .respondsWith( new Command() {
                     @Override
                     public void execute() {
-                        final UploadForm uploadForm = iocManager.lookupBean( UploadForm.class ).getInstance();
-                        //When pop-up is closed destroy bean to avoid memory leak
-                        uploadForm.addCloseHandler( new CloseHandler<PopupPanel>() {
-
-                            @Override
-                            public void onClose( CloseEvent<PopupPanel> event ) {
-                                iocManager.destroyBean( uploadForm );
-                            }
-
-                        } );
-                        uploadForm.show();
+                        uploadFormPresenter.showView();
                     }
```

Comment:
Do it one of the two ways ;)

1) Either don't ```@Inject``` it, but use ```iocManager``` to create an instance "on the fly" and then destroy it in the ```onClose``` (as per the removed code), or

2) Use ```@Inject``` and have a ```reset()``` method (or something on the Presenter) to erm, reset it -- or even have the ```showView()``` reset the form - assuming if you're showing the form you want it "reset" anyway.